### PR TITLE
RELATED: RAIL-135 Support for creating project

### PR DESCRIPTION
**Pull request by vojty**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/209/ : SUCCESS in 1m 55s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/210/ : SUCCESS in 1m 16s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/212/ : FAILURE in 53s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/213/ : SUCCESS in 1m 15s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/214/ : SUCCESS in 52s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/215/ : SUCCESS in 56s

File: src/metadata.js [hunk 1]
```diff
@@ -542,3 +542,72 @@ export function getValidElements(projectId, id, options = {}) {
 export function deleteObject(uri) {
     return del(uri);
 }
+
```
Comment (vojty) [inline src/metadata.js]: missing doc

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/216/ : SUCCESS in 3m 10s

File: src/metadata.js [hunk 2]
```diff
@@ -542,3 +542,80 @@ export function getValidElements(projectId, id, options = {}) {
...
+    return post(`/gdc/md/${projectId}/ldm/manage2`, {
+        data: JSON.stringify({
+            manage: { maql }
+        })
+    })
+    .then(parseJSON)
+    .then((response) => {
+        const manageStatusUri = response.entries[0].link;
+        return handlePolling(manageStatusUri, (task) => {
+            return isTaskFinished(task);
```
Comment (vaclavbohac) [inline src/metadata.js]: `isTaskFinished` can be passed directly

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/217/ : SUCCESS in 51s

File: src/metadata.js [hunk 3]
```diff
@@ -542,3 +542,80 @@ export function getValidElements(projectId, id, options = {}) {
...
+/**
+ * ETL pull
+ *
+ * @experimental
+ * @method etlPull
+ * @param {String} projectId
+ * @param {String} uploadsDir
+ * @param {Object} options for polling (maxAttempts, pollStep)
+ */
+
```
Comment (vaclavbohac) [inline src/metadata.js]: Remove the empty line

File: src/metadata.js [hunk 4]
```diff
@@ -542,3 +542,80 @@ export function getValidElements(projectId, id, options = {}) {
...
+export function etlPull(projectId, uploadsDir, options = {}) {
+    return post(`/gdc/md/${projectId}/etl/pull2`, {
+        data: JSON.stringify({
+            pullIntegration: uploadsDir
+        })
+    })
+    .then(parseJSON)
+    .then((response) => {
+        const etlPullStatusUri = response.pull2Task.links.poll;
+        return handlePolling(etlPullStatusUri, (task) => {
```
Comment (vaclavbohac) [inline src/metadata.js]: `isTaskFinished` can be passed directly

File: src/util.js [hunk 5]
```diff
@@ -18,3 +20,32 @@ import { get } from 'lodash';
  * @private
  */
 export const getIn = path => object => get(object, path);
+
+/**
+ * Helper for polling
+ *
+ * @param {String} uri
+ * @param {Function} pollCheck TODO better name
```
Comment (vaclavbohac) [inline src/util.js]: Forgotten TODO?

File: src/util.js [hunk 6]
```diff
@@ -18,3 +20,32 @@ import { get } from 'lodash';
...
+ * @private
+ */
+export const handlePolling = (uri, pollCheck, options = {}) => {
```
Comment (vaclavbohac) [inline src/util.js]: Document options?

File: test/metadata_test.js [hunk 7]
```diff
@@ -32,6 +33,148 @@ describe('metadata', () => {
...
+                function mockIntialPost() {
+                    fetchMock.mock(
+                        '/gdc/md/1/ldm/manage2',
+                        'POST',
+                        {
+                            status: 200,
+                            body: JSON.stringify(triggerLdmManageResponse)
+                        }
+                    );
+                }
```
Comment (vaclavbohac) [inline test/metadata_test.js]: I'd add space between the fn and `it`

File: test/metadata_test.js [hunk 8]
```diff
@@ -32,6 +33,148 @@ describe('metadata', () => {
...
+            const runningTask = mockTask('RUNNING');
+            const manageStatusUri = '/gdc/md/1/tasks/1/status';
+            const triggerLdmManageResponse = {
+                entries: [
+                    {
+                        link: manageStatusUri
+                    }
+                ]
+            };
+            describe('/gdc/md/1/ldm/manage2 call successful', () => {
```
Comment (vaclavbohac) [inline test/metadata_test.js]: I'd add space between the object definition and `describe`

File: test/project_test.js [hunk 9]
```diff
@@ -262,21 +263,8 @@ describe('project', () => {
...
-                            counter += 1;
-                            const response = counter > 3 ? createdProject : pendingProject;
-
-                            return {
-                                status: 200,
-                                body: JSON.stringify(response)
-                            };
-                        }
-                    );
+                    mockPollingRequest(projectUri, pendingProject, createdProject);
```
Comment (vaclavbohac) [inline test/project_test.js]: 👍

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/218/ : SUCCESS in 51s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/219/ : SUCCESS in 50s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/221/ : SUCCESS in 54s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/2/ : SUCCESS in 5s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/223/ : SUCCESS in 1m 36s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-js-merge-compliance-check/26/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-el7/224/ : SUCCESS in 2m 07s

**Pull request merged**